process_report/invoices/billable_invoice.py

```diff
@@ -115,6 +116,31 @@ def export_s3(self, s3_bucket):
         super().export_s3(s3_bucket)
         s3_bucket.upload_file(self.old_pi_filepath, self.PI_S3_FILEPATH)
 
+    def _filter_partners(self, data):
+        active_partnerships = list()
+        institute_list = util.load_institute_list()
+        for institute_info in institute_list:
+            if partnership_start_date := institute_info.get(
+                "mghpcc_partnership_start_date", None
+            ):
+                if (
+                    util.get_month_diff(self.invoice_month, partnership_start_date[:-3])
```

Comment:
Usually I'd add a comment when you need to access some very specific indexes of a string. 

In this case, however, I think we should convert it to a datetime object and then get the month and year. 
```
>>> "{:%Y-%m}".format(datetime.strptime("2024-09-01", "%Y-%m-%d"))
'2024-09'
```
This would also work in the case where someone writes the date as "2024-9-1" in institute_list.yaml because nothing is enforcing that a zero is added.

```
>>> "{:%Y-%m}".format(datetime.strptime("2024-9-1", "%Y-%m-%d"))
'2024-09'
```
as opposed to

```
>>> "2024-9-1"[:-3]
'2024-'
```